Makefile

```diff
@@ -411,7 +411,7 @@ start-controller-local:
 
 .PHONY: hack-serve-docs
 hack-serve-docs: hack-build-dev-tools
-	$(CONTAINER_RUNTIME) run $(DOCKER_OPTS) -p $(DOCS_PORT):$(DOCS_PORT) kargo:dev-tools make serve-docs
+	$(CONTAINER_RUNTIME) run $(DOCKER_OPTS) --user $(id -u):$(id -g) -p $(DOCS_PORT):$(DOCS_PORT) kargo:dev-tools make serve-docs
```

Comment:
What paths are problematic, as we have a fix elsewhere to address this for certain other paths that get mounted: https://github.com/akuity/kargo/blob/main/Dockerfile.dev#L27-L33

Reply:
what's the purpose of Dockerfile.dev and how is it related in this context?

Reply:
It's the Dockerfile for an image with both Go and Node and anything else needed to support development tasks (if it isn't or cannot be covered by the tool installation targets.)

Reply:
Ok, what should we do in order to support the above changes?

Reply:
What are the steps to reproduce the issue? @hiddeco asked what paths were problematic.

Reply:
All I did was run the `make hack-serve-docs` command that resulted in the following error
```
 EACCES  EACCES: permission denied, open '/workspaces/kargo/docs/_tmp_61_f8a007a35b8c63922ca99b1fba2579a9'
```
This was happening because the user inside the container doesn't have enough permissions to write to the directory. Hence, I modified the command in the Makefile to make sure we run as the host-user.

Reply:
The file/dir it's complaining about isn't a Kargo thing. What even is it?

Reply:
@nitishfy did you ever get to the bottom of this?